### PR TITLE
Implementation of the Vet Appointments API

### PR DESCRIPTION
**Pull request by rskinnerc**
#### Specifically, in this PR we have:
- [x] Setup and configured a Rails application
- [x] Setup Rubocop as linter and fixed linter errors
- [x] Setup a Postgresql DB
- [x] Setup RSpec for testing
- [x] Added the appointments' API routes and controllers (index and create)
- [x] Configured ActiveStorage to support doctor's photo upload
- [x] Added the doctor's API routes and controllers (index, create and delete)
- [x] Added a user's authentication API endpoint to handle username login
- [x] Created the user, appointment, and doctor models and DB migrations
- [x] Wrote tests for models and requests
- [x] Implemented the API documentation using RSwag (you can access the documentation at `/api-docs`)
- [x] Setup Rails CORS Gem

**NOTE** Please be aware that we have implemented the backend and frontend parts in two different repositories. Please find the [Frontend PR](https://github.com/rskinnerc/vet-appointments-react/pull/21) here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,106 @@
...
 
-### Prerequisites
+and navigate to the cloned directory:
 
-### Setup
+```
+cd vet-appointments-rails
+```
 
 ### Install
```
Comment (Meltrust) [inline README.md:38]: - [ ] On your readme, please add the required step `rails db:create` which is missing so your user can run your app. 💪

File: README.md [hunk 2]
```diff
@@ -1,56 +1,106 @@
 ![](https://img.shields.io/badge/Microverse-blueviolet)
 
-# Project Name
-
-> Description the project.
+# Vet-Appointments App
```
Comment (Meltrust) [inline README.md:3]: - [ ] A screenshot for this API project would be nice.  A shot of the rswag doc would be great, for instance. 

That way, your repo will look more appealing 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @rskinnerc,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  App is working ✔️ 
-  Good readme ✔️ 
-  Nice API docs ✔️ 

You can do it

## Suggested Changes ♻️
_Check the comments under the review._

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.


Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------

File: README.md (hunk 2 above)
Comment (rskinnerc) [reply README.md:3]: Dear @Meltrust thank you for your recommendations, we have implemented the changes you proposed and also improved some styling.

**Review by Meltrust — APPROVED** [review]
Hi STUDENT_MENTION,


## Wow, you did it 🎉
Brilliant
## Thank you for the changes you implemented 💪 🥇 ㊗️ 
### Unless you want to add more features, please go ahead to your final presentation.
### Well done, and good luck!





Congratulations! 🎉 

### To highlight

- Beautiful project✔️ 
- Nice documentation✔️ 
- The app works well✔️ 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

You can use as many of my suggestions as you want.  If there is anything you would like to skip - feel free to do that.  However, I strongly recommend you take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏


------

**Pull request merged**